### DISCUSSION TITLE
Help with securityprofile

### DISCUSSION THREAD

**Discussion by ruitabordaext** [citrineos/citrineos#12] [Q&A] [answered]
Hello all,

I'm using your centrineos-core to test our implementation of ocpp and also the everest ecosystem. Do you support securityprofile 2 and 3? What do we have to do to be able to test CentrineOs securityprofile 2 and 3 with everest.

Thanks for the help in advance

**Comment by thanaParis** [comment, marked as answer]
Hi @ruitabordaext !
We do support security profiles 2 & 3, via adjustments to the system config file.

[See here](https://github.com/citrineos/citrineos-core/blob/3825ddab60f05646139d4b5ffbb0fb9e425f0a45/00_Base/src/config/types.ts#L116) the schema for websocket security and websocket servers.

[This is the system config file used by docker compose](https://github.com/citrineos/citrineos-core/blob/3825ddab60f05646139d4b5ffbb0fb9e425f0a45/Server/src/config/envs/docker.ts). You would need to add the websocket security component and websocket servers with security profiles 2 and 3.

Here is an example of what that would look like:

```     
        websocketSecurity: {
            tlsKeysFilepath: "./data/csms_root_rsa.key",
            tlsCertificateChainFilepath: "./data/csms_chain_rsa.crt",
            mtlsCertificateAuthorityRootsFilepath: "./data/charging_station_rsa.crt",
            mtlsCertificateAuthorityKeysFilepath: "./data/charging_station_rsa.key"
        },
        websocketServer: [
            {
                securityProfile: 0,
                host: "0.0.0.0",
                port: 8081,
                protocol: "ocpp2.0.1"
            },
            {
                securityProfile: 1,
                host: "0.0.0.0",
                port: 8082,
                protocol: "ocpp2.0.1"
            },
            {
                securityProfile: 2,
                host: "0.0.0.0",
                port: 8443,
                protocol: "ocpp2.0.1"
            },
            {
                securityProfile: 3,
                host: "0.0.0.0",
                port: 8444,
                protocol: "ocpp2.0.1"
            }
        ]

```

You would need to generate the necessary key and crt files yourself (using a tool such as openssl or via a Certificate Authority) and place them in the Server/data folder (or you could adjust the paths in the above config to look elsewhere). You can then use OCPP messages to update the charger, via InstallCertificate or a Triggered SignCertificate.

**Reply by ruitabordaext** [reply]
Thank you very much for the quick response. This is a great help!
